react-app/src/components/Extractor.js

```diff
@@ -0,0 +1,312 @@
+                    onChange={(e) => {
+                      const newArgs = [...tempArgs];
+                      const currentArg = newArgs.find((temp) => arg.key === temp.key);
+                      currentArg[arg.key] = e.target.value;
+                      updateArgData(newArgs, false);
+                    }}
+                    placeholder="Enter URL"
+                  />
+                </div>
+              </Form.Group>
+            );
+          case 'dropdown':
+            return (
+              <Form.Group className="mb-3" controlId={arg.key} key={arg.key}>
+                <div>
+                  <Form.Label>{arg.label}</Form.Label>
+                </div>
+                <Form.Text>
+                  Enter field(s) as a list separated by a commas. The options are gender, mrn, name, address, birthDate,
+                  language, ethnicity, birthsex, and race.
+                </Form.Text>
+                <div className="label-and-icon-container">
+                  <Form.Control
+                    type="text"
+                    value={arg[arg.key]}
+                    onChange={(e) => {
+                      const newArgs = [...tempArgs];
+                      const currentArg = newArgs.find((temp) => arg.key === temp.key);
+                      currentArg[arg.key] = e.target.value;
+                      updateArgData(newArgs, false);
+                    }}
+                    className="arg-input-width-limit"
+                  />
+                  <Trash2
+                    onClick={() => {
+                      const newArgs = [...tempArgs];
+                      const currentArg = newArgs.find((temp) => arg.key === temp.key);
+                      currentArg.included = false;
+                      updateArgData(newArgs, false);
+                    }}
+                    className="mouse-pointer"
+                  />
+                </div>
+              </Form.Group>
+            );
+          default:
+            return (
+              <p key={arg.key}>
+                This is a placeholder for unidentified argument {i}: {arg.key}
+              </p>
+            );
+        }
+      });
+  }
+
+  function updateArgs(inputArgs, isAccordion = true) {
+    let tempArgs;
+    if (isAccordion === true) {
+      tempArgs = [...args];
+    } else {
+      tempArgs = [...inputArgs];
+    }
+    setArgsJSX(getArgsJSX(tempArgs, updateArgs));
+    if (isAccordion !== true) {
+      const constructorArgs = {};
+      tempArgs
+        .filter((arg) => arg.included === true)
+        .forEach((arg) => {
+          constructorArgs[arg.key] = arg[arg.key];
+        });
+      setArgs(tempArgs);
+      props.onArgsChange(constructorArgs);
+    }
+  }
+
+  function addArg(eventKey) {
+    const tempArgs = [...args];
+    const currentArg = tempArgs.find((arg) => arg.key === eventKey);
+    currentArg.included = true;
+    updateArgs(tempArgs, false);
+  }
+
+  function getHasArgs() {
```

Comment:
This function looks like it's just checking if an extractor has argument to add. Can we simplify the name and just call it `hasArgs`?

Reply:
I have changed this function's name to hasArgs.